### PR TITLE
[13.0][MIG] account_credit_control: Migration to 13.0

### PR DESCRIPTION
**Pull request by manuelcalerosolis**
This is the migration to v 13.0 module account_credit_control

@Tecnativa TT21599

### CODE REVIEW

File: account_credit_control/models/credit_control_line.py
Comment (ernestotejeda) [inline account_credit_control/models/credit_control_line.py:282]: I think you can simply remove `view_type`, without adding `binding_view_types`

File: account_credit_control/views/credit_control_line.xml
Comment (ernestotejeda) [inline account_credit_control/views/credit_control_line.xml:261]: I think you can simply remove `view_type`, without adding `binding_view_types`

File: account_credit_control/views/credit_control_policy.xml
Comment (ernestotejeda) [inline account_credit_control/views/credit_control_policy.xml:94]: I think you can simply remove `view_type`, without adding `binding_view_types`

File: account_credit_control/data/data.xml
Comment (ernestotejeda) [inline account_credit_control/data/data.xml]: There are several places in the module where the model appears first and then the ID when declaring a record in XML, we could take advantage now to fix that according to the OCA conventions:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#21format

**Review by ernestotejeda — COMMENTED** [review]
Thanks @manuelcalerosolis , Good job !!!!

File: account_credit_control/models/credit_control_line.py
Comment (manuelcalerosolis) [reply account_credit_control/models/credit_control_line.py:282]: thanks @ernestotejeda

File: account_credit_control/views/credit_control_policy.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/credit_control_policy.xml:94]: thanks @ernestotejeda

File: account_credit_control/models/credit_control_policy.py
Comment (ernestotejeda) [inline account_credit_control/models/credit_control_policy.py]:
```suggestion
        help="For policies which should not generate lines or are obsolete"
```
Search for " " occurrences throughout the code and remove them in unnecessary cases like this, there are about five.

File: account_credit_control/models/credit_control_line.py
Comment (ernestotejeda) [inline account_credit_control/models/credit_control_line.py:165]:
```suggestion
                "invoice_id": move_line.move_id.id,
```

File: account_credit_control/models/account_move.py
Comment (ernestotejeda) [inline account_credit_control/models/account_move.py:25]: Will account_credit_control.group_account_credit_control_info be enough?

File: account_credit_control/views/account_move.xml
Comment (ernestotejeda) [inline account_credit_control/views/account_move.xml]: Will account_credit_control.group_account_credit_control_user be enough?

File: account_credit_control/views/credit_control_line.xml
Comment (ernestotejeda) [inline account_credit_control/views/credit_control_line.xml]: Will account_credit_control.group_account_credit_control_info be enough?

File: account_credit_control/views/credit_control_policy.xml
Comment (ernestotejeda) [inline account_credit_control/views/credit_control_policy.xml]: Will account_credit_control.group_account_credit_control_user be enough?

File: account_credit_control/views/credit_control_run.xml
Comment (ernestotejeda) [inline account_credit_control/views/credit_control_run.xml]: Will account_credit_control.group_account_credit_control_user be enough?

File: account_credit_control/views/res_partner.xml
Comment (ernestotejeda) [inline account_credit_control/views/res_partner.xml]: Will account_credit_control.group_account_credit_control_user be enough?

File: account_credit_control/wizard/credit_control_policy_changer.py
Comment (ernestotejeda) [inline account_credit_control/wizard/credit_control_policy_changer.py:36]:
```suggestion
            if invoice.type != "out_invoice":
```
Other types of journal entries that are not sales invoices are also those that are type 'entry', 'in_receipt', 'out_receipt'.

File: account_credit_control/models/credit_control_policy.py
Comment (ernestotejeda) [inline account_credit_control/models/credit_control_policy.py]:
```suggestion
        return self._move_lines_subset(controlling_date, "account.move", "move_id")
```

**Review by ernestotejeda — CHANGES_REQUESTED** [review]

File: account_credit_control/views/credit_control_line.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/credit_control_line.xml:261]: Fixed!

File: account_credit_control/models/account_move.py
Comment (manuelcalerosolis) [reply account_credit_control/models/account_move.py:25]: It is the same as in the previous version

File: account_credit_control/views/credit_control_line.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/credit_control_line.xml]: It is the same as in the previous version

File: account_credit_control/views/credit_control_run.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/credit_control_run.xml]: It is the same as in the previous version

File: account_credit_control/wizard/credit_control_policy_changer.py
Comment (manuelcalerosolis) [reply account_credit_control/wizard/credit_control_policy_changer.py:36]: Good observation

File: account_credit_control/views/credit_control_policy.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/credit_control_policy.xml]: the same of previous version

File: account_credit_control/views/res_partner.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/res_partner.xml]: the same of previous version

File: account_credit_control/views/account_move.xml
Comment (manuelcalerosolis) [reply account_credit_control/views/account_move.xml]: the same of previous version

File: account_credit_control/views/credit_control_line.xml
Comment (CasVissers-360ERP) [inline account_credit_control/views/credit_control_line.xml]: In EE this will create a new menu named "invoicing"

File: account_credit_control/report/report_credit_control_summary.xml
Comment (CasVissers-360ERP) [inline account_credit_control/report/report_credit_control_summary.xml]: Number is not a field of account.move, name should be sufficient?

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Initial feedback

File: account_credit_control/views/credit_control_line.xml
Comment (pedrobaeza) [reply account_credit_control/views/credit_control_line.xml]: We can't do anything about this. Odoo decided to include EE menu only and relocate all the entries. The only solution is a glue module for fixing this.

File: account_credit_control/wizard/credit_control_policy_changer.py
Comment (ernestotejeda) [reply account_credit_control/wizard/credit_control_policy_changer.py:36]: @manuelcalerosolis , this conversation was marked as resolved, however I don't see this change has been done.

File: account_credit_control/wizard/credit_control_communication.py
Comment (CasVissers-360ERP) [inline account_credit_control/wizard/credit_control_communication.py]: Field name is changed to store_fname in v13?

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Incorrect field on ir.attachment found

File: account_credit_control/models/credit_control_run.py
Comment (carlosdauden) [inline account_credit_control/models/credit_control_run.py:17]: Add this:
```suggestion
    _order = "date DESC"

```
Or cherry-pick this:
https://github.com/OCA/credit-control/pull/61/files#diff-dee1c1533bb5904fd05aa4b7c6685c5cR15

**Comment by rafaelbn** [comment]
Please don't merge until my review. Thanks

File: account_credit_control/security/account_security.xml
Comment (CasVissers-360ERP) [inline account_credit_control/security/account_security.xml]: All record rules should be updated to ['|',('company_id','=',False),('company_id', 'in', company_ids)]

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Record rules are incorrect.

File: account_credit_control/security/account_security.xml
Comment (pedrobaeza) [reply account_credit_control/security/account_security.xml]: Fixed record rules and added guideline in migration guide:

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0/_compare/bc329d0741d5606c7b4d441c64cd46e727da5528...fbcfc75d7f45d3057044c6943df9f4a94c8f652f

File: account_credit_control/wizard/credit_control_communication.py
Comment (pedrobaeza) [reply account_credit_control/wizard/credit_control_communication.py]: Changed, thanks

**Comment by pedrobaeza** [comment]
@CasVissers @rafaelbn can you please re-review?

File: account_credit_control/report/report_credit_control_summary.xml
Comment (CasVissers-360ERP) [reply account_credit_control/report/report_credit_control_summary.xml]: @pedrobaeza 
I still get:
```
  File "/Users/brahoo/PycharmProjects/cust/odoo-13.0/odoo/addons/base/models/ir_qweb.py", line 363, in _get_field
    field = record._fields[field_name]
KeyError: 'number'

Error to render compiling AST
KeyError: 'number'
Template: account_credit_control.report_credit_control_summary_document
Path: /t/t/div/table/tbody/tr/t[1]/td/span
Node: <span t-field="l.invoice_id.number"/>
```

I think the line <span t-field="l.invoice_id.number"/> should be removed. Only name is sufficient.

**Comment by CasVissers-360ERP** [comment]
https://drive.google.com/file/d/1piDHuSzMDWha06kQoXrDMh89bijioySg/view

Even though a credit control line has been processed it seems that it is created a 2nd time? Or am I doing something wrong?

**Comment by pedrobaeza** [comment]
Is that happening the same on v12?

**Comment by CasVissers-360ERP** [comment]
No I can't reproduce the same behavior in v12. I think it's wrong any way we look at it?

**Comment by pedrobaeza** [comment]
If this doesn't happen in v12, then indeed it's a problem of the migration. I will take over this PR and check that. Thanks for reviewing.

**Comment by CasVissers-360ERP** [comment]
@pedrobaeza 
any progress?

**Comment by pedrobaeza** [comment]
Sorry, not yet. I will try ASAP.

**Comment by bhaveshselarka** [comment]
Hello Team,

Thanks for this PR and peoples are involved in this PR.
it's a very useful module for community versions.

Is it possible to merge in the main branch? 
@pedrobaeza @chienandalu @manuelcalerosolis 

once again thanks.

Regards,

**Comment by bhaveshselarka** [comment]
@OCA/accounting-maintainers any chance to merge?

**Comment by bhaveshselarka** [comment]
Hello Team,

Thanks for this PR and peoples are involved in this PR.
it's a very useful module for community versions.

Is it possible to merge in the main 13.0 branch?
@pedrobaeza @chienandalu @manuelcalerosolis @CasVissers @rafaelbn

once again thanks.

Regards,

File: account_credit_control/models/credit_control_policy.py
Comment (CasVissers-360ERP) [inline account_credit_control/models/credit_control_policy.py]: @pedrobaeza 
Finally found the error.. Should be `AND (NOT manually_overridden OR manually_overridden IS NULL))`

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Fix found for level not increasing

**Comment by CasVissers-360ERP** [comment]
@bhaveshselarka 
Maybe if you stop spamming it will go a bit faster.

File: account_credit_control/models/credit_control_policy.py
Comment (pedrobaeza) [reply account_credit_control/models/credit_control_policy.py]: This seems something specific of your DB, as boolean fields shouldn't contain nulls... Any way, I think it can be put as `AND manually_overridden IS NOT True` for being shorter.

Comment (pedrobaeza) [reply account_credit_control/models/credit_control_policy.py]: I have changed it anyway in the latest change

**Comment by pedrobaeza** [comment]
Let's finally merge it:

/ocabot merge nobump

File: account_credit_control/models/credit_control_policy.py
Comment (CasVissers-360ERP) [reply account_credit_control/models/credit_control_policy.py]: Not sure but it was finding newly or not yet vacuumed records. Where the value was null. 

Was not specific to my DB, same issue was present in runbot as I showed in my video.

Anyhow thanks for the fix!

Comment (pedrobaeza) [reply account_credit_control/models/credit_control_policy.py]: OK, it's weird, but this happens due to this SQL legacy code. Everything should come through ORM... Next thing to improve!

Thanks for the investigation.

**Comment by CasVissers-360ERP** [comment]
@pedrobaeza 
Maybe I missed something here..

Is  <record id="sales_journal" model="account.journal">  and   <record id="a_sale" model="account.account"> not demo data? They are now created when installing/updating the module.

**Comment by pedrobaeza** [comment]
I don't get you.

**Comment by CasVissers-360ERP** [comment]
```
<record id="a_sale" model="account.account">
        <field name="code">X2020</field>
        <field name="name">Product Sales - (test)</field>
        <field name="user_type_id" ref="account.data_account_type_revenue" />
        <field name="tag_ids" eval="[(6,0,[ref('account.account_tag_operating')])]" />
    </record>
    <!-- sales journal -->
    <record id="sales_journal" model="account.journal">
        <field name="name">Customer Invoices - Test</field>
        <field name="code">TINV</field>
        <field name="type">sale</field>
        <field name="default_credit_account_id" ref="a_sale" />
        <field name="default_debit_account_id" ref="a_sale" />
        <field name="refund_sequence" eval="True" />
    </record>
```

Is in data.xml,   "data/data.xml",  is in manifest? I think this is demo data?
image

**Comment by pedrobaeza** [comment]
Yes, I don't know why they are there. Can you please do a PR amending it?

**Pull request merged**
